ray-operator/controllers/ray/utils/httpproxy_httpclient.go

```diff
@@ -57,11 +57,17 @@ func (r *RayHttpProxyClient) CheckProxyActorHealth(ctx context.Context) error {
 	if err != nil {
 		return err
 	}
-	defer resp.Body.Close()
+	defer func() {
+		_, _ = io.Copy(io.Discard, resp.Body)
```

Comment:
Do we need to call `io.Copy(io.Discard, resp.Body)` after `io.ReadAll(resp.Body)` for responses that are not `http.StatusOK`? If not, can you call `io.Copy(io.Discard, resp.Body)` only for `http.StatusOK`?

```go
if resp.StatusCode != http.StatusOK {
  ...
} else {
  // Add a comment to explain ...
  _, _ = io.Copy(io.Discard, resp.Body)
}
```